### ISSUE TITLE
sc2128 with parameter expansions

### ISSUE THREAD

**Issue by izabera** [koalaman/shellcheck#542]
these don't trigger sc2128 but they should
```bash
arr=(x y z)
echo "${arr#x}" "${arr##x}" "${arr%x}" "${arr%%x}"
```
etc...
